core/src/machine/ppu.rs

```diff
@@ -184,20 +193,30 @@ impl Ppu {
     /// function panics!
     pub(crate) fn store_io_byte(&mut self, addr: Word, byte: Byte) {
         match addr.get() {
-            0xFF40 => self.lcd_control = byte,
+            0xFF40 => {
+                let was_enabled = self.lcd_enabled();
+                self.lcd_control = byte;
+                match (was_enabled, self.lcd_enabled()) {
+                    (false, true) => {
+                        info!("[ppu] LCD was enabled");
+                        self.set_phase(Phase::OamSearch);
+                        self.cycle_in_line = 0;
+                        // TODO: also reset other stuff?
```

Comment:
Do we need to reset the current line here too? (e.g. `self.current_line = Byte::new(0);`)

Reply:
I don't think so. The current line is set to 0 a few lines below, when the LCD is disabled. I actually think that most things should be "reset" when the LCD is disabled instead of when it is enabled. 

I think the TODO is a bit outdated. But it might still be useful until we figured this out.